aws/resource_aws_appsync_resolver.go

```diff
@@ -0,0 +1,182 @@
+	}
+
+	mutexKey := fmt.Sprintf("appsync-schema-%s", d.Get("api_id").(string))
+	awsMutexKV.Lock(mutexKey)
+	defer awsMutexKV.Unlock(mutexKey)
+
+	_, err := retryOnAwsCode(appsync.ErrCodeConcurrentModificationException, func() (interface{}, error) {
+		return conn.CreateResolver(input)
+	})
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId(d.Get("api_id").(string) + "-" + d.Get("type").(string) + "-" + d.Get("field").(string))
+
+	return resourceAwsAppsyncResolverRead(d, meta)
+}
+
+func resourceAwsAppsyncResolverRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).appsyncconn
+
+	apiID, typeName, fieldName, err := decodeAppsyncResolverID(d.Id())
+
+	if err != nil {
+		return err
+	}
+
+	input := &appsync.GetResolverInput{
+		ApiId:     aws.String(apiID),
+		TypeName:  aws.String(typeName),
+		FieldName: aws.String(fieldName),
+	}
+
+	resp, err := conn.GetResolver(input)
+	if err != nil {
+		return err
+	}
+
+	d.Set("api_id", apiID)
+	d.Set("arn", resp.Resolver.ResolverArn)
+	d.Set("type", resp.Resolver.TypeName)
+	d.Set("field", resp.Resolver.FieldName)
+	d.Set("data_source", resp.Resolver.DataSourceName)
+	d.Set("request_template", resp.Resolver.RequestMappingTemplate)
+	d.Set("response_template", resp.Resolver.ResponseMappingTemplate)
+
+	return nil
+}
+
+func resourceAwsAppsyncResolverUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).appsyncconn
+
+	input := &appsync.UpdateResolverInput{
+		ApiId:                   aws.String(d.Get("api_id").(string)),
+		DataSourceName:          aws.String(d.Get("data_source").(string)),
+		FieldName:               aws.String(d.Get("field").(string)),
+		TypeName:                aws.String(d.Get("type").(string)),
+		RequestMappingTemplate:  aws.String(d.Get("request_template").(string)),
+		ResponseMappingTemplate: aws.String(d.Get("response_template").(string)),
+	}
+
+	mutexKey := fmt.Sprintf("appsync-schema-%s", d.Get("api_id").(string))
+	awsMutexKV.Lock(mutexKey)
+	defer awsMutexKV.Unlock(mutexKey)
+
+	_, err := retryOnAwsCode(appsync.ErrCodeConcurrentModificationException, func() (interface{}, error) {
+		return conn.UpdateResolver(input)
+	})
+
+	if err != nil {
+		return err
+	}
+
+	return resourceAwsAppsyncResolverRead(d, meta)
+}
+
+func resourceAwsAppsyncResolverDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).appsyncconn
+
+	apiID, typeName, fieldName, err := decodeAppsyncResolverID(d.Id())
+
+	if err != nil {
+		return err
+	}
+
+	input := &appsync.DeleteResolverInput{
+		ApiId:     aws.String(apiID),
+		TypeName:  aws.String(typeName),
+		FieldName: aws.String(fieldName),
+	}
+
+	mutexKey := fmt.Sprintf("appsync-schema-%s", d.Get("api_id").(string))
+	awsMutexKV.Lock(mutexKey)
+	defer awsMutexKV.Unlock(mutexKey)
+
+	_, err = retryOnAwsCode(appsync.ErrCodeConcurrentModificationException, func() (interface{}, error) {
+		return conn.DeleteResolver(input)
+	})
+
+	if err != nil {
+		return err
```

Comment:
We should return error context here for operators and code maintainers:

```suggestion
		return fmt.Errorf("error deleting AppSync Resolver (%s): %s", d.Id(), err)
```